### ISSUE TITLE
Unhelpful error message for invalid polymer.json

### ISSUE THREAD

**Issue by tony19** [Polymer/polymer-cli#354]
### Description
When `polymer.json` contains invalid JSON (e.g., from a trailing comma on array element), `polymer-cli` emits a generic warning about `polymer.json` and continues to build, which leads to other build errors.

```
$ polymer build
warn:    Could not load config file 
{ path: 'polymer.json', err: {} }
...
```

This message is logged at [project-config.ts:40](https://github.com/Polymer/polymer-cli/blob/master/src/project-config.ts#L40), and this could be improved by storing the error message itself (instead of the error object) in the log metadata:
```
- logger.warn('Could not load config file', {path: filepath, err: error});
+ logger.warn('Could not load config file', {path: filepath, err: error.message});
```

This emits:

```
$ polymer build
warn:    Could not load config file 
{ path: 'polymer.json', err: 'Unexpected token ] in JSON at position 248' }
...
```

Further, the invalid JSON should fail the build immediately.

### Versions & Environment
- Polymer CLI: 0.14.0
- node: 6.3.1
- Operating System: OS X El Capitan

#### Steps to Reproduce

1. Create a project: `polymer init app-drawer-template`
2. In `polymer.json`, add a trailing comma to an array element. Or copy the following into `polymer.json`:

  ```json
{
  "entrypoint": "index.html",
  "shell": "src/my-app.html",
  "fragments": [
    "src/my-view1.html",
    "src/my-view2.html",
    "src/my-view3.html",
  ]
}
```
3. Build: `polymer build`
4. Observe the obscure warning (and subsequent build error).

#### Expected Results
Fail-fast build error with specific error message

Example:
```
$ polymer build
error:    Could not load config file 
{ path: 'polymer.json', err: 'Unexpected token ] in JSON at position 156' }
```

#### Actual Results
```
$ polymer build
warn:    Could not load config file 
{ path: 'polymer.json', err: {} }
... continues to build, leading to red herrings ...
```

**Comment by garcianavalon** [comment]
+1 I was having that same unhelpful output. The error was  a trailing coma in last item, fixing it made warn disappear ^_^

**Comment by FredKSchott** [comment]
@tony19 thanks for the great issue description and for taking the time to think through this. I completely agree, we should exit if there's an error parsing the `polymer.json` file (but still continue if the file is just not found).

Would anyone like to write the fix & a test for this?

**Comment by tony19** [comment]
@FredKSchott Ok, done. Thanks

**Comment by FredKSchott** [comment]
Fixed via #375
